model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -339,14 +339,16 @@ message MonitoringInfoSpecs {
   enum Enum {
     // TODO(ajamato): Add the PTRANSFORM name as a required label after
     // upgrading the python SDK.
-    USER_COUNTER = 0 [(monitoring_info_spec) = {
+    USER_METRIC = 0 [(monitoring_info_spec) = {
```

Comment:
beam:metric:user: is a prefix, not a full URN, and probably shouldn't be in this table.

Reply:
Currently, metrics are either of "user" type (defined by {ptransform, namespace, name}), or they're one of the 5 "system"-metric types below. 

It's true that the "user" category is not a "specific URN", so there is tension with e.g. [this docstring above](https://github.com/apache/beam/pull/7868/files#diff-4d226ebf3f70cf18449c29e82511a67eR333).

I think of this table as enumerating the possible metrics-URN *categories* (indeed, that's what it does, and what our design calls for, as opposed to e.g. enumerating all allowed "user"-metric URNs here). The 5 "system" categories are each inhabited by exactly one URN, so they are particularly easy to deal with in SDKs, while the user-metric category requires special treatment (which, conveniently, the SDKs already implement).

In this PR, I am laying some groundwork toward supporting more "types" of metrics (both "user" and "system"). "User" metrics should be able to take on the three types listed here (counter, distribution, gauge).

With that context, there's a few directions I can see taking your comment:
- split the "user" category into 3 type-specific user-categories
- leave as-is, but make the docs more clear about the special status of `USER_METRIC`'s "spec"
- some larger overhaul involving removing user metrics from this table.

Let me know what you think!

Reply:
The "user metric" prefix should not be in this table. (We also shouldn't be restricting user metrics here, any valid metric type is a valid user metric type.)

@ajamato

Reply:
OK. Should [`USER_METRIC_URN_PREFIX`](https://github.com/apache/beam/pull/7868/files#diff-4d226ebf3f70cf18449c29e82511a67eR466) (aka `USER_COUNTER_URN_PREFIX`) also not be in `MonitoringInfoUrns.Enum`, or is the issue just with this `MonitoringInfoSpecs` table?

Reply:
Just an issue with the table, which should be listing specific metrics.

Reply:
@Ardagan and I have discussed that its not really the best idea to make the URN a prefix for the user metric. As this has led to writing a lot of code to parse the URN to obtain those fields out. And being this weird exception makes all the code have to special case it.

It would be better to package the namespace and name as a label on the MonitoringInfo.
Then we could use the same URN everywhere, no parsing, no special casing.

Given this, then there are two choices i.e.
(1) Use a different URN for user gauge, counter and distribution
- "beam:metric:user:counter:v1"
- "beam:metric:user:gauge:v1"
- "beam:metric:user:distribution:v1"
Then change the MonitoringInfoSpec:
      required_labels: [ "PTRANSFORM", "NAMESPACE", "NAME" ],
and each will have a separate  type_urn in their spec.
      type_urn:


(2) Use a single URN "beam:metric:user:v1" for user gauge, counter, distribution
Then change the MonitoringInfoSpec as well:
      required_labels: [ "PTRANSFORM", "NAMESPACE", "NAME" ],
and we will NOT enforce the type_urn in the spec.


I prefer #1, as this will better describe each one, and if we add new user metric styles we can add a spec for them